test_runner/regress/test_compute_metrics.py

```diff
@@ -1,21 +1,146 @@
+        if queries is not None:
+            for query in queries:
+                log.info("Checking query %s in %s", query["query_name"], collector_config)
+                endpoint.safe_psql(query["query"])
+
+    endpoint.stop_and_destroy()
+
+
+SQL_EXPORTER = shutil.which("sql_exporter")
+
+
+@pytest.mark.skipif(SQL_EXPORTER is None, reason="sql_exporter is not in the PATH")
+@pytest.mark.parametrize(
+    "sql_exporter_config,collector_config",
+    [
+        ("sql_exporter.yml", "neon_collector.yml"),
+        ("sql_exporter_autoscaling.yml", "neon_collector_autoscaling.yml"),
+    ],
+)
+def test_sql_exporter_metrics_e2e(
+    neon_simple_env: NeonEnv,
+    monkeypatch: MonkeyPatch,
+    tmp_path: Path,
+    sql_exporter_config: str,
+    collector_config: str,
+):
     """
-    Test compute metrics, exposed in the neon_backend_perf_counters and
-    neon_perf_counters views
+    This is a full E2E test of the sql_exporter setup to make sure it works
+    without error.
     """
     env = neon_simple_env
-    endpoint = env.endpoints.create_start("main")
-
-    conn = endpoint.connect()
-    cur = conn.cursor()
-
-    # We don't check that the values make sense, this is just a very
-    # basic check that the server doesn't crash or something like that.
-    #
-    # 1.5 is the minimum version to contain these views.
-    cur.execute("CREATE EXTENSION neon VERSION '1.5'")
-    cur.execute("SELECT * FROM neon_perf_counters")
-    cur.execute("SELECT * FROM neon_backend_perf_counters")
+
+    endpoint = env.endpoints.create("main")
+    endpoint.respec(skip_pg_catalog_updates=False)
+    endpoint.start()
+
+    # Point the config at our endpoint
+    new_sql_exporter_config = str(tmp_path / "sql_exporter.yml")
+    with (
+        open(
+            CONFIG_DIR / sql_exporter_config,
+            "r",
+            encoding="utf-8",
+        ) as i,
+        open(new_sql_exporter_config, "w", encoding="utf-8") as o,
+    ):
+        conn_options = endpoint.conn_options()
+        host: str = conn_options["host"]
+        port: str = conn_options["port"]
+
+        config: dict[Any, Any] = yaml.safe_load(i)
+        config["target"][
+            "data_source_name"
+        ] = f"postgresql://cloud_admin@{host}:{port}/postgres?sslmode=disable&application_name=sql_exporter"
+        config["collector_files"] = [str(CONFIG_DIR / collector_config)]
+        yaml.dump(config, o)
+
+    # sql_exporter will fail to launch if this environment variable exists
+    monkeypatch.delenv("PGSERVICEFILE")
+
+    sql_exporter = subprocess.Popen(
+        [
+            cast("str", SQL_EXPORTER),
+            "-config.file",
+            new_sql_exporter_config,
+            "-web.listen-address",
+            ":9499",
+        ],
+    )
+
+    # Give time for sql_exporter to launch
+    time.sleep(3)
+
+    resp = requests.get("http://localhost:9499/metrics")
+    sql_exporter.kill()
+    resp.raise_for_status()
+
+    endpoint.stop_and_destroy()
```

Comment:
AFAIR, we have a cleanup part in `env`, no need to do it explicitly in the test it self